### PR TITLE
Update Envoy to v1.21.0

### PR DESCRIPTION
**Pull request by shuoyang2016**

### CODE REVIEW

File: .bazelversion [hunk 1]
```diff
@@ -1 +1 @@
-4.1.0
+4.2.0
```
Comment (nareddyt) [inline .bazelversion:1]: Match envoy

https://github.com/envoyproxy/envoy/blob/main/.bazelversion#L1

**Review by qiwzhang — APPROVED** [review]

**Review by TAOXUY — APPROVED** [review]

**Review by nareddyt — APPROVED** [review]

**Pull request merged**